tests/apidoc_login_test.go

```diff
@@ -81,6 +82,70 @@ func TestLogin(t *testing.T) {
 				},
 			})
 		})
+		// sytest: POST /login can log in as a user with just the local part of the id
+		t.Run("POST /login can log in as a user with just the local part of the id", func(t *testing.T) {
+			t.Parallel()
+			createDummyUser(t, unauthedClient, "@test_user:matrix")
+			reg := regexp.MustCompile("@([^:]*):")
+			localPart := reg.FindStringSubmatch("@test_user:matrix")
+			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+				"type": "m.login.password",
+				"identifier": {
+					"type": "m.id.user",
+					"user": "`+localPart[1]+`"
```

Comment:
Just use `"user": "test_user"`

Reply:
I refactored the creatDummyUser to return userID so we may mimic the similar scenario of attests.